src/interface/lazy.jl

```diff
@@ -103,12 +113,12 @@ function Base.map(f, src::LazyTensor, args...)
     T = return_type(DefaultAlgebra(), f, eltype(src), map(eltype, args)...)
     new_default = f(map(fill_value, largs)...)
     data = mapjoin(immediate(f), ldatas...)
-    return LazyTensor{T}(identify(data), src.extrude, new_default)
+    return LazyTensor{T}(identify(data), src.extrude, src.shape, new_default)
 end
 
 function Base.map!(dst, f, src::LazyTensor, args...)
     res = map(f, src, args...)
-    return LazyTensor(identify(reformat(dst, res.data)), res.extrude, res.fill_value)
+    return LazyTensor(identify(reformat(dst, res.data)), res.extrude, res.size, res.fill_value)
```

Comment:
Almost certainly, this should be `size(res)` or `res.shape`, depending on whether or not we know that `res` is a `LazyTensor`.